addon/ng2/models/watcher.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+var chalk   = require('chalk');
+var Task    = require('ember-cli/lib/models/task');
+var debug   = require('debug')('angular-cli:watcher');
+var Promise = require('ember-cli/lib/ext/promise');
+var exec    = Promise.denodeify(require('child_process').exec);
+var isWin = /^win/.test(process.platform);
+
+var Watcher = Task.extend({
+  verbose: true,
+
+  init: function() {
+    var options = this.buildOptions();
+
+    debug('initialize %o', options);
+
+    this.watcher = this.watcher || new (require('broccoli-sane-watcher'))(this.builder, options);
+
+    this.watcher.on('error', this.didError.bind(this));
+    this.watcher.on('change', this.didChange.bind(this));
+  },
+
+  didError: function(error) {
+    debug('didError %o', error);
+    this.ui.writeError(error);
+    this.analytics.trackError({
+      description: error && error.message
+    });
+  },
+
+  then: function() {
+    return this.watcher.then.apply(this.watcher, arguments);
+  },
+
+  didChange: function(results) {
+    debug('didChange %o', results);
+    var totalTime = results.totalTime / 1e6;
+
+    this.ui.writeLine('');
+    this.ui.writeLine(chalk.green('Build successful - ' + Math.round(totalTime) + 'ms.'));
+
+    this.analytics.track({
+      name:    'angular rebuild',
+      message: 'broccoli rebuild time: ' + totalTime + 'ms'
+    });
+
+    this.analytics.trackTiming({
+      category: 'rebuild',
+      variable: 'rebuild time',
+      label:    'broccoli rebuild time',
+      value:    parseInt(totalTime, 10)
+    });
+    
+    if (this.options.buildCompleted) {
+      return this.options.buildCompleted();
+    }
+  },
+
+  on: function() {
+    this.watcher.on.apply(this.watcher, arguments);
+  },
+
+  off: function() {
+    this.watcher.off.apply(this.watcher, arguments);
+  },
+  buildOptions: function() {
```

Comment:
Is the whole file a copy of the ember one? Maybe we can just import it from there then.